src/Files.App/DataModels/SidebarPinnedModel.cs

```diff
@@ -280,6 +282,20 @@ public async Task AddItemToSidebarAsync(string path)
 			AddLocationItemToSidebar(locationItem);
 		}
 
+		private async Task<byte[]?> RetrieveItemIconData(string itemPath, FilesystemResult<BaseStorageFolder> result)
+		{
+			byte[]? iconData = null;
+
+			if (result)
```

Comment:
Can you check for and retrieve result in the calling function and not this one?

Reply:
Could do it, yet by doing so we remove a security in this function. May be a problem if it's called somewhere else in the future. I'm doing the modification nonetheless.